docs/tileentities/tileentity.md

```diff
@@ -0,0 +1,128 @@
+There might be situations in which you need to change your `BlockState`, an example for this is the vanilla furnace,
+which changes its state from `lit=false` to `lit=true` when fuel and something smetlable is inside.
+Achieving this is rather simple, by overriding following method
+```JAVA
+    TileEntity#shouldRefresh(World world, BlockPos pos, IBlockState oldState, IBlockState newSate)#boolean
+```
+!!! note
+    You should actually check the `BlockStates` and not just return `false` in order to prevent unwanted behavior and bugs.
+    
+## Ticking `TileEntities`
+If you need a ticking `TileEntity` for example to keep track of the progress during a smelting process.
+You need to add the `ITickable` interface to your `TileEntity`.
+Now you can implement all your calculations within
+```JAVA
+    ITickable#update()
+```
+!!! note
+    This method is called each tick, therefore you should avoid having complicated calculations in here.
+    If possible, you should make more complex calculations just every X ticks.
+    (The amount of ticks in a second may be lower then 20 (twenty) but won't be higher)
+    
+## Synchronizing the Data to the Client
+There are 3 (three) ways of syncing data to the client.
+Synchronizing on chunk load, synchronizing on block updates and synchronizing with a custom network message.
+
+### Synchronizing on chunk load
+For this you need to override
+```JAVA
+    TileEntity#getUpdateTag()
+
+    TileEntity#handleUpdateTag(NBTTagCompound nbt)
+```
+Again, this is pretty simple, the first method collects the data that should be send to the client.
+While the second one processes that data. If your `TileEntity` doesn't contain much data you might be able to use the methods out of the `Storing Data within your TileEntity` section.
+!!! note
+    Synchronizing excessive/useless data for TileEntities can lead to network congestion. You should optimize your network usage by sending only the information the client needs when the client needs it. For instance, it is more often that not unnecessary to send the inventory of a tile entity in the update tag, as this can be synchronized via its GUI.
+
+### Synchronizing on block update
+This method is a bit more complicated, but again you just need to override 2 methods.
+Here is a tiny example implementation of it 
+```JAVA
+    @Override
+    public SPacketUpdateTileEntity getUpdatePacket(){
+        NBTTagCompound nbtTag = new NBTTagCompound();
+        //Write your data into the nbtTag
+        return new SPacketUpdateTileEntity(getPos(), 1, nbtTag);
+    }
+
+    @Override
+    public void onDataPacket(NetworkManager net, SPacketUpdateTileEntity pkt){
+        NBTTagCompound tag = pkt.getNbtCompound();
+        //Handle your Data
+    }
+```
+The Constructor of `SPacketUpdateTileEntity` takes:
+The position of your `TileEntity`.
+An ID, though it isn't really used besides by Vanilla, therefore you can just put a 1 in there.
+An `NBTTagCompound` which should contain your data.
```

Comment:
These are not going to be on separate lines on RTD, so it will look like the following:

The Constructor of `SPacketUpdateTileEntity` takes: The position of your `TileEntity`. An ID, though it isn't really used besides by Vanilla, therefore you can just put a 1 in there. An `NBTTagCompound` which should contain your data.

Consider using a list.